weblate/api/views.py

```diff
@@ -927,6 +928,36 @@ def units(self, request, **kwargs):
 
         return self.get_paginated_response(serializer.data)
 
+    @action(detail=True, methods=["post"])
+    def autotranslate(self, request, **kwargs):
+        translation = self.get_object()
+        project = translation.component.project
+        if not request.user.has_perm("translation.auto", project):
```

Comment:
The permission check should be at translation level:

```suggestion
        if not request.user.has_perm("translation.auto", translation):
```